pkg/api/validate.go

```diff
@@ -98,7 +98,17 @@ func (p *Properties) validateVMProfile(vmconf VMConfigurator) error {
 	if e := validateOSDisk(vm.OSDisk); e != nil {
 		return e
 	}
-
+	if(vm.SecurityProfile == nil) {
```

Comment:
validate() is not supposed to change the values, but to validate their correctness.
Use default.go to modify and set defaults.

Reply:
Sure, moved the defaults' part to defaults.go, in the new iteration. And we set SecurityProfile to nil if both the under-lying strings are "none"